File: csharp/AdminEmployee.cs
using System;
using System.Text.Json.Serialization;

[Serializable]
public class AdminEmployee : FullTimeEmployee
{
    [JsonInclude]
    private string Clearance { get; set; }

    public AdminEmployee() : base() { }

    public AdminEmployee(string id, string firstName, string lastName, DateTime dob,
        string position, string department, double baseSalary, string benefits, string clearance)
        : base(id, firstName, lastName, dob, position, department, baseSalary, benefits)
    {
        if (department != "Administration")
            throw new InvalidDataException("AdminEmployee must belong to the Administration department.");
        Clearance = clearance;
    }

    public override void ReportToManager()
    {
        if (Manager != null)
            Console.WriteLine($"{GetFullName()} (Admin Employee, clearance: {Clearance}) reports to manager {Manager.GetFullName()}");
        else
            Console.WriteLine($"{GetFullName()} (Admin Employee, clearance: {Clearance}) has no assigned manager.");
    }

    public void ShowClearance() => Console.WriteLine($"{GetFullName()}'s clearance level: {Clearance}");
}

File: java/AdminEmployee.java
import java.util.Date;

public class AdminEmployee extends FullTimeEmployee {
    private String clearance;

    public AdminEmployee(String id, String firstName, String lastName, Date dob,
            String position, String department, double baseSalary, String benefits,
            String clearance) throws InvalidDataException {
        super(id, firstName, lastName, dob, position, department, baseSalary, benefits);

        // Subclass-specific validation
        if (!"Administration".equals(department)) {
            throw new InvalidDataException("AdminEmployee must belong to the Administration department.");
        }

        this.clearance = clearance;
    }

    @Override
    public void reportToManager() {
        if (manager != null) {
            System.out.println(getFullName() + " (Admin Employee, clearance: " + clearance + ") reports to manager "
                    + manager.getFullName());
        } else {
            System.out.println(
                    getFullName() + " (Admin Employee, clearance: " + clearance + ") has no assigned manager.");
        }
    }

    public void showClearance() {
        System.out.println(getFullName() + "'s clearance level: " + clearance);
    }
}